### DISCUSSION TITLE
Client side fetching vs Incremental static generation

### DISCUSSION THREAD

**Discussion by doptster** [vercel/next.js#17715] [Help] [answered]
Greetings

Let's say I have a list of popular topics to be display on my landing page from API, which will be updated every few days.
In this scenario, client side fetching or getstaticprops (with revalidate) is more doable? 

My main concern on the selection would be performance and SEO friendly, does choosing any of them makes any difference? I am also worrying that choosing the incremental static generation approach might be overkill for the application.

**Comment by lfades** [comment, marked as answer]
I recommend reading the [Static Generation](https://nextjs.org/docs/basic-features/pages#static-generation-recommended) section in our docs first, and then our section for [Fetching data on the client side](https://nextjs.org/docs/basic-features/data-fetching#fetching-data-on-the-client-side).

Both are SEO friendly, but Static generation will give you better results in that regard, and also provide a faster experience and more responsive for all of your visitors. In general I recommend to use Static Generation for public data, and client side rendering for data that depends on a cookie and/or is related to a specific visitor and where you feel comfortable showing a loading state.

[Incremental Static Generation](https://nextjs.org/docs/basic-features/data-fetching#incremental-static-regeneration) is the same Static Generation but with the additional step that you can now have a fallback and use it to create new static pages after the initial build (without having to build the app again)

We also have [Incremental Static Regeneration](https://nextjs.org/docs/basic-features/data-fetching#incremental-static-regeneration), which is the same Static Generation but allows you to update already generated pages in the background.

Hope that helps 🙏

**Reply by doptster** [reply]
alright, thanks for the recommendation.
